packages/flutter_tools/lib/src/flx.dart

```diff
@@ -29,14 +29,15 @@ const String _kSnapshotKey = 'snapshot_blob.bin';
 Future<int> createSnapshot({
   String mainPath,
   String snapshotPath,
-  String depfilePath
+  String depfilePath,
+  String packages
 }) {
   assert(mainPath != null);
   assert(snapshotPath != null);
 
   final List<String> args = <String>[
     tools.getHostToolPath(HostTool.SkySnapshot),
-    '--packages=${path.absolute(PackageMap.globalPackagesPath)}',
+    '--packages=${packages ?? path.absolute(PackageMap.globalPackagesPath)}',
```

Comment:
Rather than generalizing this function, I think we should create a `build_mojo.dart` command and have it call the snapshotter directly.  That way we can avoid having to mess with the `tools` static.

After it invokes the snapshotter, it can use `assemble` to make the FLX with all the assets and such.